### PR TITLE
[stable/sonarqube] Option for settings encryption secret

### LINKED ISSUE #10675: [stable/sonarqube] Add secrets
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Feature Request

**Version of Helm and Kubernetes**: 2.12.1, 1.10

**Which chart**: stable/sonarqube

**What happened**: I cannot specify a mapping for secrets for env vars and I cannot specify a secret to source sonarqube.properties from

**What you expected to happen**: I would like a way to store secrets (such as ldap password) and reference them from the chart

**Anything else we need to know**:

I can open a PR with secretEnvs or similar properties if necessary, but I wanted to start the conversation first.

Thank you!

### PR DESCRIPTION
**Pull request by jcoeltjen**
Add another option to use a Kubernetes Secret for settings
encryption by setting the the sonar.secretKeyPath property
and mounting the actual secret file as a volume.

Signed-off-by: Jannis Oeltjen <oss@jcoeltjen.de>

#### What this PR does / why we need it:
This PR adds another option to be able to set the secret used by the [Settings Encryption](https://docs.sonarqube.org/latest/instance-administration/security/#header-6) feature to a value stored in a Kubernetes Secret.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #10675

#### Special notes for your reviewer:
Tested on AKS for both cases of having custom `sonar.properties` set and without this option set.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by jcoeltjen** [comment]
Rebased PR because of conflicting changes in master.

File: stable/sonarqube/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: sonarqube
 description: Sonarqube is an open sourced code quality scanning tool
-version: 0.13.5
-appVersion: 7.6
+version: 0.14.0
+appVersion: 7.4
```
Comment (rjkernick) [inline stable/sonarqube/Chart.yaml]: Please update the appVersion to 7.6 to reflect the recent change

**Review by rjkernick — CHANGES_REQUESTED** [review]

File: stable/sonarqube/Chart.yaml (hunk 1 above)
Comment (jcoeltjen) [reply stable/sonarqube/Chart.yaml]: I fixed the wrong rebase. 
The appVersion should now be unchanged by this PR.

**Comment by jcoeltjen** [comment]
Is there anything I need to to before this PR can be merged?

**Comment by rjkernick** [comment]
Looks like you need to bump the version in chart.yaml once more. Thanks for the patience. There has been a lot of activity on this chart in the last week.

**Comment by jcoeltjen** [comment]
> Looks like you need to bump the version in chart.yaml once more. Thanks for the patience. There has been a lot of activity on this chart in the last week.

Will do that. Can you have a look at #11444 first.
Then I don't need to bump twice.

**Pull request merged**